src/shipit_uplift/shipit_uplift/coverage_by_changeset_impl.py

```diff
@@ -0,0 +1,196 @@
+import json
+import requests
+import whatthepatch
+from config import CoverageService, COVERAGE_SERVICE
+
+
+github_commits = {}
+def get_github_commit(mercurial_commit):
+    if mercurial_commit not in github_commits:
+        url = 'https://api.pub.build.mozilla.org/mapper/gecko-dev/rev/hg/%s'
+        r = requests.get(url % mercurial_commit)
+
+        github_commits[mercurial_commit] = r.text.split(' ')[0]
+
+    return github_commits[mercurial_commit]
+
+
+coverage_data = {}
+def get_coverage(changeset, filename):
+    if COVERAGE_SERVICE == CoverageService.COVERALLS:
+        r = requests.get('https://coveralls.io/builds/%s/source.json' % get_github_commit(changeset), params={
+            'filename': filename,
+        })
+
+        if r.status_code != requests.codes.ok:
+            return None
+
+        return r.json()
+    elif COVERAGE_SERVICE == CoverageService.CODECOV:
+        r = requests.get('https://codecov.io/api/gh/marco-c/gecko-dev/src/%s/%s' % (get_github_commit(changeset), filename))
+
+        data = r.json()
+
+        if r.status_code != requests.codes.ok:
+            if data['error']['reason'] == 'File not found in report':
+                return None
+
+            raise Exception('Can\'t load codecov.io report')
+
+        return dict([(int(l), v) for l, v in data['commit']['report']['files'][filename]['l'].items()])
+    elif COVERAGE_SERVICE == CoverageService.ACTIVEDATA:
+        print(json.dumps({
```

Comment:
Would you mind switching the print statements to logging? (Unless there's a specific reason not to).

Reply:
I've removed them, they were only used for debugging purposes.